src/controls/dragDropFiles/DragDropFiles.tsx

```diff
@@ -0,0 +1,219 @@
+
+  constructor(props: IDragDropFilesProps) {
+    super(props);
+    // Initialize state
+    this.state = {
+      dragStatus: false
+    };
+
+  }
+
+  /**
+ * Lifecycle hook when component is mounted
+ */
+  public componentDidMount(): void {
+    const { dropEffect, enable } = this.props;
+    if (dropEffect == undefined || dropEffect == "") {
+      this._dropEffect = "copy";
+    } else {
+      this._dropEffect = dropEffect;
+    }
+    if (enable == undefined) {
+      this._enable = true;
+    } else {
+      this._enable = enable;
+    }
+    // Add EventListeners for drag zone area
+    let divDropArea = this.dropRef.current;
+    if(this._enable == true)
+    {
+      divDropArea.addEventListener('dragenter', this.handleonDragEnter);
+      divDropArea.addEventListener('dragleave', this.handleonDragLeave);
+      divDropArea.addEventListener('dragover', this.handleonDragOver);
+      divDropArea.addEventListener('drop', this.handleonDrop);
+    }
+  }
+
+  /**
+   * Stop listeners from onDragOver event.
+   * @param e
+   */
+  private handleonDragOver = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.dataTransfer.items && e.dataTransfer.items.length > 0) {
+      e.dataTransfer.dropEffect = e.dataTransfer.dropEffect = this.props.dropEffect;
+    }
+  }
+  /**
+   * Stop listeners from onDragEnter event, enable drag and drop view.
+   * @param e
+   */
+  private handleonDragEnter = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.dragCounter++;
+    if (e.dataTransfer.items && e.dataTransfer.items.length > 0) {
+      e.dataTransfer.dropEffect = this._dropEffect;
+      this.setState({ dragStatus: true });
+    }
+  }
+  /**
+   * Stop listeners from ondragenter event, disable drag and drop view.
+   * @param e
+   */
+  private handleonDragLeave = (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.dragCounter--;
+    if (this.dragCounter === 0) {
+      this.setState({ dragStatus: false });
+    }
+  }
+  /**
+  * Stop listeners from onDrop event and load files to property onDrop.
+  * @param e
+  */
+  private handleonDrop = async (e) => {
+    e.preventDefault();
+    e.stopPropagation();
+    this.setState({ dragStatus: false });
+    if (e.dataTransfer && e.dataTransfer.items) {
+      this.props.onDrop(await this.getFilesAsync(e));
+    }
+    e.dataTransfer.clearData();
+    this.dragCounter = 0;
+  }
+
+  /**
+   * Add File to Array Files of type File[]
+   * https://www.meziantou.net/upload-files-and-directories-using-an-input-drag-and-drop-or-copy-and-paste-with.htm 
+   * @param dataTransfer 
+   */
+  private getFilesAsync = async (e) => {
+    const Customfiles = e.dataTransfer;
+    const items = Customfiles.items;
+    const Directory = [];
+    const files: File[] = [];
+    for (let i = 0; i < items.length; i++) {
+        const item = items[i];
+        if (item.kind === "file") {
+            if (typeof item.webkitGetAsEntry === "function") {
```

Comment:
@aaclage could you please add some detail why webkitGetAsEntry is being used? I believe this was not used before.
Quickly checked and is listed here as experimental feature: https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/webkitGetAsEntry
Also has the following information: 
"This function is implemented as webkitGetAsEntry() in non-WebKit browsers including Firefox at this time; it may be renamed to getAsEntry() in the future, so you should code defensively, looking for both."

Reply:
@joelfmrodrigues This was included to read folders with files inside, since is not yet fully supported i will remove it. just going to read files.

Reply:
@joelfmrodrigues would be ok if we include a validation to retrieve files from Folders, i think will be a big value to have this option.
Out of the box sharePoint uses List Drag and drop and with a defensive method also:
if (item.kind === "file") {
          if (item.getAsEntry)
          {
            entry = item.getAsEntry();
            Directory.push(entry);
          }else  if (item.webkitGetAsEntry){
            entry = item.webkitGetAsEntry();
            Directory.push(entry);
          }else if ("function" == typeof item.getAsFile) {
            const file = item.getAsFile();
              if (file) {
                files.push(file);
              }
          }
          continue;
      }